core/src/main/scala/akka/persistence/cassandra/journal/TagWriters.scala

```diff
@@ -295,19 +296,15 @@ import akka.util.ByteString
         val startTime = System.nanoTime()
 
         def writeTagScanningBatch(group: Seq[(String, Long)]): Future[Done] = {
-          val batch = new BatchStatement(BatchStatement.Type.UNLOGGED)
-          group.foreach {
-            case (pid, seqNr) => batch.add(ps.bind(pid, seqNr: JLong))
+          val statements: Seq[BoundStatement] = group.map {
+            case (pid, seqNr) => ps.bind(pid, seqNr: JLong)
           }
-          tagWriterSession.executeStatement(batch)
+          Future.traverse(statements)(tagWriterSession.executeStatement).map(_ => Done)
         }
 
-        // Group the updates into 500 statements per UNLOGGED BatchStatement. These
-        // are executed sequentially to not induce too much load that might influence
-        // performance of other writes and reads. See issue #408.
-        // The size of the data is small and fixed so no need to configure the batch size.
+        // Execute 10 async statements at a time to not introduce too much load see issue #408.
         val batchIterator: Iterator[Future[Done]] =
-          updates.grouped(500).map(writeTagScanningBatch)
+          updates.grouped(10).map(writeTagScanningBatch)
```

Comment:
Is this really 10 at a time and not all in chunks of 10 executed at the same time?

Reply:
yeah you're right, stupid!